pkg/block/s3/adapter.go

```diff
@@ -793,6 +787,9 @@ func (a *Adapter) RuntimeStats() map[string]string {
 	}
 }
 
+// extractS3Server extracts the responding server from the response.
```

Comment:
what about this one?

Reply:
added custom middleware to put and upload part to support this functionality.